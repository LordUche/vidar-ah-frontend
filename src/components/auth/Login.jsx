import React from 'react';

const Login = () => (
  <>
    <h1>Welcome to The Login Page</h1>
  </>
);

export default Login;
