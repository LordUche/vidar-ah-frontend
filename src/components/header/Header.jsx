import React from 'react';

export default function Listing() {
  return <div>Hello World</div>;
}
