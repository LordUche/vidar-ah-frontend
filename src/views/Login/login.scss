@import '../../assets/scss/settings';

// .form-login {
//   font-style: $font-stack;
//   font-size: .9em;
//   color: red;
// }

.yellow-button-login {
  width: $input-field-width;

&:focus {
  outline: none;
}
}

.container-extras {
  // border: 1px solid white;
  display: flex;
  font-size: 1em;
  font-style: $font-stack;
  justify-content: space-between;
  margin: 20px auto;
  width: $input-field-width;

  & span {
    display: block;
    font-size: 1em;
     font-style: $font-stack;

    & .checkbox {
      height: 1em;
      margin-right: .5em;
      width: 1em;
    }

    & .link-to-another-page {
      color: #ffa500;
    font-size: 1em;
       font-style: $font-stack;

    }

    & .link-to-another-page:hover {
      color: #ffffff;
    }

  }
}

.container-directing-to-signup {
  // border: 1px solid white;
  margin: 6em 0 2em 0;

  & span {
    display: block;
    font-size: 1em;
    font-style: $font-stack;

    & .link-to-another-page {
      color: #ffa500;
      font-size: 1em;
      font-style: $font-stack;
      margin: 0 1em 0 1em;
    }
  }
}
