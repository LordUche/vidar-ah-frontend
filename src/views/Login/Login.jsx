import React from 'react';
import {
  Container,
  Row,
  Col,
  Form
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import Header from '../../components/header/Header';
import Button from '../../components/button/Button';

const Login = ({ history }) => (
  <>
    <div className="purple-gradient">
      <Container>
        <Header location={history.location.pathname} />
      </Container>
      <Container>
        <Row>
          <Col md={{ span: 6, offset: 3 }} className="form-black-bg">
            <h3>Login</h3>
            <Form>
              <Form.Control type="email" placeholder="Email" className="dark-forms" />
              <Form.Control type="password" placeholder="Password" className="dark-forms" />
              <Button text="Login" />
            </Form>
          </Col>
        </Row>
      </Container>
    </div>
  </>
);

Login.propTypes = {
  history: PropTypes.string.isRequired
};
export default Login;
