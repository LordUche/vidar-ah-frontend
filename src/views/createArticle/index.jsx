/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Container, Alert } from 'react-bootstrap';
import Create from '../../components/createForms/Create';
import Preview from '../../components/previewArticle/Preview';
import { validateArticle } from '../../utils/validator';
import { createArticle } from '../../redux/actions/articles';
import Header from '../../components/header/Header';
import './createArticle.scss';

function CreateArticle(props) {
  const [articleData, setArticleData] = useState({
    title: null,
    description: null,
    body: null
  });
  const [component, setComponent] = useState('write');
  const [loading, setLoading] = useState(false);
  const [errorsArray, setErrors] = useState([]);
  const [successMessage, setSuccessMessage] = useState('');

  async function completeArticleCreation() {
    setLoading(true);
    setErrors([]);
    const res = await props.createArticle(articleData);
    const { errors } = res;
    setLoading(false);
    if (errors) {
      setErrors(errors);
    }
    setSuccessMessage(res.message);
    props.history.push(`/articles/${res.article.slug}`);
  }

  function runValidators(e) {
    e.preventDefault();
    const errors = validateArticle(articleData);
    if (errors.length > 0) {
      setComponent('write');
      return setErrors(errors);
    }
    return completeArticleCreation();
  }

  function updateInput(e) {
    if (!(e.target.name === 'images')) {
      return setArticleData({ ...articleData, [e.target.name]: e.target.value });
    }
    return setArticleData({ ...articleData, images: e.target.files[0] });
  }

  function deleteArticle() {
    window.location = '/create-article';
  }

  function renderErrors(errors) {
    return errors.map(error => (
      <Alert key={error} variant="danger">{error}</Alert>
    ));
  }

  function renderSuccess(message) {
    return message ? <Alert variant="success">{message}</Alert> : null;
  }

  return (
    <div className="create-article-container">
      <Container>
        <Header type="purple" />
        {renderErrors(errorsArray)}
        {renderSuccess(successMessage)}
      </Container>
      <div>
        {
          component === 'write'
            ? (
              <Create
                onChangeText={updateInput}
                onSubmit={() => setComponent('preview')}
                onDelete={deleteArticle}
                article={articleData}
              />
            )
            : (
              <Preview
                onSubmit={runValidators}
                onDelete={deleteArticle}
                article={articleData}
                continueEdit={() => setComponent('write')}
                loading={loading}
              />
            )
        }
      </div>
    </div>
  );
}


CreateArticle.propTypes = {
  createArticle: PropTypes.func.isRequired
};

export default connect(() => ({}), { createArticle })(CreateArticle);
