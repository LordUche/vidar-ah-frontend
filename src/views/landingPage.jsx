import React from 'react';
import Header from '../components/header/Header';

export default function LandingPage() {
  return <div><Header /></div>;
}
