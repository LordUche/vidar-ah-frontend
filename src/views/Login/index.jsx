import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { SocialIcon } from 'react-social-icons';
import {
  Container, Row, Col, Form
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import Header from '../../components/header/Header';
import Button from '../../components/button/Button';
import { loginUser } from '../../redux/actions/auth';
import Loader from '../../components/loader/Loader';
import './login.scss';

export const Login = ({ history, loginUser: handleLogin, loading }) => {
  const [userCredentials, setUserCredentials] = useState({
    email: null,
    password: null
  });

  const updateLocalState = (e) => {
    setUserCredentials({ ...userCredentials, [e.target.name]: e.target.value });
  };

  const onSubmit = async (e) => {
    e.preventDefault();
    const successfulLogin = await handleLogin(userCredentials);
    if (successfulLogin) {
      history.push('/');
    }
  };

  return (
    <>
      <div className="purple-gradient-bg">
        {loading && <Loader />}
        <Container>
          <Header location={history.location.pathname} />
        </Container>
        <Container>
          <Row>
            <Col md={{ span: 6, offset: 3 }} className="form-black-bg">
              <h3>Login</h3>
              <Form onSubmit={e => onSubmit(e)} className="form-login">
                <Form.Control
                  type="email"
                  name="email"
                  placeholder="Email"
                  className="dark-forms"
                  required
                  onChange={e => updateLocalState(e)}
                />
                <Form.Control
                  type="password"
                  name="password"
                  placeholder="Password"
                  className="dark-forms"
                  required
                  onChange={e => updateLocalState(e)}
                />
                <Button text="Login" />
                <div className="container-extras">
                  <span>
                    <input
                      type="checkbox"
                      name="rememberMe"
                      className="checkbox"
                    />
                    Remember Me
                  </span>
                  <span>
                    <NavLink
                      to="/requestpasswordreset"
                      className="link-to-another-page"
                    >
                      Forgotten password?
                    </NavLink>
                  </span>
                </div>
              </Form>
              <div className="container-social-login">
                <span className="info">Or login using </span>
                <SocialIcon
                  url="http://twitter.com"
                  className="social-media-icons"
                />
                <SocialIcon
                  url="http://facebook.com"
                  className="social-media-icons"
                />
                <SocialIcon
                  url="http://google.com"
                  className="social-media-icons"
                />
              </div>
              <div className="container-directing-to-signup">
                <span>
                  Don't have an account? Singn up
                  <NavLink to="/signup" className="link-to-another-page">
                     here
                  </NavLink>
                </span>
              </div>
            </Col>
          </Row>
        </Container>
      </div>
    </>
  );
};

Login.propTypes = {
  history: PropTypes.object.isRequired,
  loginUser: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  loading: state.fetchReducer.fetching
});

export default connect(
  mapStateToProps,
  { loginUser }
)(Login);
