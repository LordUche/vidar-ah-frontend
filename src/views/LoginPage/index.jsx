/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { SocialIcon } from 'react-social-icons';
import {
  Container, Row, Col, Form, Alert
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import Header from '../../components/header/index';
import Button from '../../components/button/Button';
import { loginUser } from '../../redux/actions/auth';
import Loader from '../../components/loader/Loader';
import './login.scss';


export const Login = ({ history, loginUser: handleLogin, loading }) => {
  const [userCredentials, setUserCredentials] = useState({
    email: null,
    password: null
  });
  const [errors, setErrors] = useState([]);

  function updateLocalState(e) {
    setUserCredentials({ ...userCredentials, [e.target.name]: e.target.value });
  }

  function renderErrors(errorsArray) {
    return errorsArray.map(error => <Alert key={error} variant="danger">{error}</Alert>);
  }

  async function onSubmit(e) {
    e.preventDefault();
    const loginRes = await handleLogin(userCredentials);
    if (!Array.isArray(loginRes)) {
      return history.push('/');
    }
    console.log(loginRes);
    return setErrors(loginRes);
  }

  return (
    <>
      <div className="purple-gradient-bg">
        {loading && <Loader />}
        <Container>
          <Header location={history.location.pathname} />
        </Container>
        <Container>
          {renderErrors(errors)}
          <Row>
            <Col md={{ span: 6, offset: 3 }} className="form-black-bg">
              <h3>Login</h3>
              <Form onSubmit={onSubmit} className="form-login">
                <Form.Control
                  type="email"
                  name="email"
                  placeholder="Email"
                  className="dark-forms"
                  required
                  onChange={updateLocalState}
                />
                <Form.Control
                  type="password"
                  name="password"
                  placeholder="Password"
                  className="dark-forms"
                  required
                  onChange={updateLocalState}
                />
                <Button text="Login" className="yellow-button-login" onClick={() => onSubmit} />
                <div className="container-extras">
                  <span>
                    <NavLink
                      to="/requestpasswordreset"
                      className="link-to-another-page"
                    >
                      Forgotten password?
                    </NavLink>
                  </span>
                </div>
              </Form>
              <div className="container-social-login">
                <span className="info">Or login using </span>
                <SocialIcon
                  url="http://twitter.com"
                  className="social-media-icons"
                />
                <SocialIcon
                  url="http://facebook.com"
                  className="social-media-icons"
                />
                <SocialIcon
                  url="http://google.com"
                  className="social-media-icons"
                />
              </div>
              <div className="container-directing-to-signup">
                <span>
                  Don't have an account? Singn up
                  <NavLink to="/signup" className="link-to-another-page">
                    here
                  </NavLink>
                </span>
              </div>
            </Col>
          </Row>
        </Container>
      </div>
    </>
  );
};

Login.propTypes = {
  history: PropTypes.object.isRequired,
  loginUser: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  loading: state.fetchReducer.fetching
});

const LoginPage = connect(
  mapStateToProps,
  { loginUser }
)(Login);

export default LoginPage;
