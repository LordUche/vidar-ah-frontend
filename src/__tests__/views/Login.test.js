/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Login } from '../../views/Login';
import Header from '../../components/header/Header';

const getLoginPage = () => {
  const props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: ''
      }
    },

    loginUser: jest.fn(),
    loading: false
  };

  const enzymeWrapper = shallow(<Login {...props} />);
  // console.log(enzymeWrapper.debug());
  return enzymeWrapper;
};

const LoginPage = getLoginPage();
describe('Test the Login Page', () => {
  it('it render the longin page without failing', () => {
    expect(toJson(LoginPage)).toMatchSnapshot();
  });
  it('should contain the Header', () => {
    expect(LoginPage.find(Header)).toHaveLength(1);
  });
  it('it render social media icons', () => {
    expect(LoginPage.find('SocialIcon')).toHaveLength(3);
  });
});
