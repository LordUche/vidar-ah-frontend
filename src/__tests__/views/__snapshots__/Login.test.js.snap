// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Login Page it render the longin page without failing 1`] = `
<Fragment>
  <div
    className="purple-gradient-bg"
  >
    <Container
      as="div"
      fluid={false}
    >
      <Header
        location=""
      />
    </Container>
    <Container
      as="div"
      fluid={false}
    >
      <Bootstrap(Row)>
        <Col
          as="div"
          className="form-black-bg"
          md={
            Object {
              "offset": 3,
              "span": 6,
            }
          }
        >
          <h3>
            Login
          </h3>
          <Form
            as="form"
            className="form-login"
            inline={false}
            onSubmit={[Function]}
          >
            <FormControl
              as="input"
              className="dark-forms"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
            />
            <FormControl
              as="input"
              className="dark-forms"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            <Button
              text="Login"
            />
            <div
              className="container-extras"
            >
              <span>
                <input
                  className="checkbox"
                  name="rememberMe"
                  type="checkbox"
                />
                Remember Me
              </span>
              <span>
                <NavLink
                  className="link-to-another-page"
                  to="/requestpasswordreset"
                >
                  Forgotten password?
                </NavLink>
              </span>
            </div>
          </Form>
          <div
            className="container-social-login"
          >
            <span
              className="info"
            >
              Or login using 
            </span>
            <SocialIcon
              className="social-media-icons"
              url="http://twitter.com"
            />
            <SocialIcon
              className="social-media-icons"
              url="http://facebook.com"
            />
            <SocialIcon
              className="social-media-icons"
              url="http://google.com"
            />
          </div>
          <div
            className="container-directing-to-signup"
          >
            <span>
              Don't have an account? Singn up
              <NavLink
                className="link-to-another-page"
                to="/signup"
              >
                here
              </NavLink>
            </span>
          </div>
        </Col>
      </Bootstrap(Row)>
    </Container>
  </div>
</Fragment>
`;
